utils/src/main/scala/za/co/absa/enceladus/utils/transformations/DeepArrayTransformations.scala

```diff
@@ -652,11 +789,79 @@ object DeepArrayTransformations {
     *
     * @param path A path to an element of a struct (e.g. company.employee.firstName)
     * @return true if a path consists only of 1 element meaning it is the leaf element of the input column path
     */
   private def isLeafElement(path: Seq[String]): Boolean = path.lengthCompare(2) < 0
+
+  /**
+    * Converts from a simple transformation to en extended transformation.
+    * In the extended expression a function for getting columns by fully qualified names is ignored.
+    *
+    * @param transformation A transformation function that takes a column and returns a column.
+    * @return An extended transformation that takes a column and a function hat returns columns from fully qualified
+    *         names and returns a column.
+    */
+  private def toExtendedTransformation(transformation: TransformFunction): ExtendedTransformFunction = {
+    (field, _) => transformation(field)
+  }
+
+  /**
+    * The method takes a field and a parent field, both fully qualified. The method does the following:
+    * <ul>
+    *   <li>If the input field is a child of the parent field the path is split into parent and child parts and these
+    *   parts are returned as a pair.</li>
+    *   <li>If the input field is not a child of the parent field the method returns a pair of empty string and
+    *   the full qualified input field name.</li>
+    * </ul>
+    *
+    * @param field       A fully qualified input field name
+    * @param parentField A parent field to split in case the input field is its child
+    * @return A pair consisting of the parent and the child field parts
+    */
+  private[transformations] def splitParentField(field: String, parentField: String): (String, String) = {
+    if (field == parentField) {
+      (field, "")
+    } else {
+      val parentFieldWithDot = if (parentField.endsWith(".")) parentField else parentField + "."
+      if (field.startsWith(parentFieldWithDot)) {
+        (parentField, field.substring(parentFieldWithDot.length))
+      } else {
+        ("", field)
+      }
+    }
+  }
+
+  /**
+    * The method takes a field and a list of parent fields, all fully qualified. The method does the following:
+    * <ul>
+    *  <li>If the input field is a child of any of the parent fields, a longest parent is selected, and the path is
+    *  split into parent and child parts and these parts are returned as a pair.</li>
+    *  <li>If the input field is not a child of any of the listed parents the method returns a pair of empty string and
+    *  the full qualified input field name.</li>
+    * </ul>
+    *
+    * @param field        A fully qualified input field name
+    * @param parentFields A parent field to split in case the input field is its child
+    * @return A pair consisting of the parent and the child field parts
+    */
+  private[transformations] def splitByLongestParent(field: String, parentFields: Seq[String]): (String, String) = {
```

Comment:
What does it mean by the longest parent? Is `alfa.beta` (9 chars) longer than `a.b.c.d` (7 chars)? One is longer, the other is deeper. I see you use length, but the tests don't cover this, so I am still curious.

Reply:
Yeah, the terminology is too fuzzy. Going to rename to `splitByDeepestParent()`. But these methods are called only for fields in a path to one field, so the longest and the deepest here are the same.  I will clarify this in the ScalaDoc comments.

E.g. if a field being transformed is `a.b.c.d`, and 'a' and 'c' are arrays, the list of arrays will contain `a`, `a.b.c` so that 'a.x' and `a.b.x` will have `a` as the parent, while `a.b.c.d.e` will have `a.b.c` as the parent.

Reply:
In the tests, there is a list `a, a.b, aa.bb` etc. So what if I keep the logic but modify it a bit to a list of `a, a.b.c, a.dddd`. Here `length` wouldn't work, right? Is this UseCase even plausible or is that just that tests are written that way?

Reply:
counting the number of `.` (dosts) seems safer right now to me

Reply:
Cool idea! This solves a more generic case.

Reply:
> In the tests, there is a list `a, a.b, aa.bb` etc. So what if I keep the logic but modify it a bit to a list of `a, a.b.c, a.dddd`. Here `length` wouldn't work, right? Is this UseCase even plausible or is that just that tests are written that way?

After thinking a bit I think counting dots and using `.length()` is equivalent since all parent path must be subpath of the input field. The test case includes a situation when the list of parent fields does not consist of a single subpath. This case is supported but should not happen in practice in array transformations.

Reply:
You are right. Wanna change it back, or keep it?

Reply:
Will change back and add a test that will fail on dot counting